File: src/main/java/com/yzj/alzassistant/controller/AiModelController.java
package com.yzj.alzassistant.controller;

import com.mybatisflex.core.paginate.Page;
import com.yzj.alzassistant.annotation.AuthCheck;
import com.yzj.alzassistant.common.BaseResponse;
import com.yzj.alzassistant.common.DeleteRequest;
import com.yzj.alzassistant.common.ResultUtils;
import com.yzj.alzassistant.constant.UserConstant;
import com.yzj.alzassistant.exception.ThrowUtils;
import com.yzj.alzassistant.model.dto.aiModel.AiModelAddRequest;
import com.yzj.alzassistant.model.dto.aiModel.AiModelQueryRequest;
import com.yzj.alzassistant.model.dto.aiModel.AiModelUpdateRequest;
import com.yzj.alzassistant.model.entity.User;
import com.yzj.alzassistant.model.vo.AiModelVO;
import com.yzj.alzassistant.service.AiModelService;
import com.yzj.alzassistant.service.UserService;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.web.bind.annotation.*;

/**
 * 大模型管理 控制层。
 *
 * @author <a href="https://github.com/yozp">yunikon</a>
 */
@RestController
@RequestMapping("/aiModel")
public class AiModelController {

    @Resource
    private AiModelService aiModelService;

    @Resource
    private UserService userService;

    /**
     * 添加大模型
     */
    @PostMapping("/add")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Long> addAiModel(@RequestBody AiModelAddRequest addRequest, HttpServletRequest request) {
        ThrowUtils.throwIf(addRequest == null, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        Long id = aiModelService.addAiModel(addRequest, loginUser);
        return ResultUtils.success(id);
    }

    /**
     * 更新大模型
     */
    @PostMapping("/update")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> updateAiModel(@RequestBody AiModelUpdateRequest updateRequest, HttpServletRequest request) {
        ThrowUtils.throwIf(updateRequest == null, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        boolean result = aiModelService.updateAiModel(updateRequest, loginUser);
        return ResultUtils.success(result);
    }

    /**
     * 删除大模型
     */
    @PostMapping("/delete")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> deleteAiModel(@RequestBody DeleteRequest deleteRequest, HttpServletRequest request) {
        ThrowUtils.throwIf(deleteRequest == null || deleteRequest.getId() == null, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        boolean result = aiModelService.deleteAiModel(deleteRequest.getId(), loginUser);
        return ResultUtils.success(result);
    }

    /**
     * 启用大模型
     */
    @PostMapping("/enable/{id}")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> enableAiModel(@PathVariable Long id, HttpServletRequest request) {
        ThrowUtils.throwIf(id == null || id <= 0, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        boolean result = aiModelService.enableAiModel(id, loginUser);
        return ResultUtils.success(result);
    }

    /**
     * 停用大模型
     */
    @PostMapping("/disable/{id}")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> disableAiModel(@PathVariable Long id, HttpServletRequest request) {
        ThrowUtils.throwIf(id == null || id <= 0, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        boolean result = aiModelService.disableAiModel(id, loginUser);
        return ResultUtils.success(result);
    }

    /**
     * 分页查询大模型
     */
    @PostMapping("/list/page")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Page<AiModelVO>> listAiModelByPage(@RequestBody AiModelQueryRequest queryRequest) {
        ThrowUtils.throwIf(queryRequest == null, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        Page<AiModelVO> page = aiModelService.listAiModelByPage(queryRequest);
        return ResultUtils.success(page);
    }

    /**
     * 根据id获取大模型详情
     */
    @GetMapping("/get")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<AiModelVO> getAiModelById(@RequestParam Long id) {
        ThrowUtils.throwIf(id == null || id <= 0, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        AiModelVO vo = aiModelService.getAiModelById(id);
        return ResultUtils.success(vo);
    }

}

File: alz-assistant-frontend/src/api/aiModelController.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request'

/** 此处后端没有提供注释 POST /aiModel/add */
export async function addAiModel(body: API.AiModelAddRequest, options?: { [key: string]: any }) {
  return request<API.BaseResponseLong>('/aiModel/add', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /aiModel/delete */
export async function deleteAiModel(body: API.DeleteRequest, options?: { [key: string]: any }) {
  return request<API.BaseResponseBoolean>('/aiModel/delete', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /aiModel/disable/${param0} */
export async function disableAiModel(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.disableAiModelParams,
  options?: { [key: string]: any }
) {
  const { id: param0, ...queryParams } = params
  return request<API.BaseResponseBoolean>(`/aiModel/disable/${param0}`, {
    method: 'POST',
    params: { ...queryParams },
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /aiModel/enable/${param0} */
export async function enableAiModel(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.enableAiModelParams,
  options?: { [key: string]: any }
) {
  const { id: param0, ...queryParams } = params
  return request<API.BaseResponseBoolean>(`/aiModel/enable/${param0}`, {
    method: 'POST',
    params: { ...queryParams },
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 GET /aiModel/get */
export async function getAiModelById(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getAiModelByIdParams,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseAiModelVO>('/aiModel/get', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /aiModel/list/page */
export async function listAiModelByPage(
  body: API.AiModelQueryRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponsePageAiModelVO>('/aiModel/list/page', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /aiModel/update */
export async function updateAiModel(
  body: API.AiModelUpdateRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseBoolean>('/aiModel/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}
